ContextDesktop/src/main/java/com/github/migi_1/Context/model/entity/IKillable.java

```diff
@@ -0,0 +1,30 @@
+package com.github.migi_1.Context.model.entity;
+
+/**
+ * Interface for everything that is killable,
+ * forces implementation of health, and being able to take damage.
+ * @author Damian
+ *
+ */
+public interface IKillable {
```

Comment:
Shouldn't this interface also have an onKilled() method?
I would then have a collection of Killables in the Environment. Then in the update method of the Environment, in the update method check whether the health of the Killable is 0 or lower, then call the onKilled() method if it's true.